syntax = "proto3";

service NumbersService {
  rpc getNumbers(NumbersRequest) returns (NumbersResponse) {}
  rpc receiveResult(ResultRequest) returns (ResultResponse) {}
}

message NumbersRequest {
  int32 num_queues = 1;
}

message NumbersResponse {
  repeated double numbers = 1;
  repeated string publishers = 4;
}

message ResultRequest {
  string user_addr = 1;
  double result = 2;
  repeated string subscribed = 3;
}

message ResultResponse {
  bool response = 1;
}
